identity-service/src/notifications/utils.js

```diff
@@ -95,6 +96,62 @@ async function calculateTrackListenMilestonesFromDiscovery(discoveryProvider) {
   return parsedListenCounts
 }
 
+/**
+ * Queries the discovery provider and returns all subscribers for each user in
+ * userIds.
+ *
+ * @param {Set<number>} userIds to fetch subscribers for
+ * @returns Object {userId: Array[subscriberIds]}
+ */
+async function bulkGetSubscribersFromDiscovery(userIds) {
+  const userSubscribersMap = {}
+  if (userIds.size === 0) {
+    return userSubscribersMap
+  }
+
+  try {
+    const { discoveryProvider } = audiusLibsWrapper.getAudiusLibs()
+    const ids = [...userIds].map((id) => encodeHashId(id))
+    const response = await axios.post(
+      `${discoveryProvider.discoveryProviderEndpoint}/v1/full/users/subscribers`,
+      { ids: ids }
+    )
+
+    // const timeout = 1 /* min */ * 60 /* sec */ * 1000 /* ms */
+    // const subscribersFromDN = await discoveryProvider.bulkGetSubscribers(
```

Comment:
Do we prefer hitting API endpoints through the libs wrapper or directly with `discoveryProvider.discoveryProviderEndpoint`? It looks like both approaches are used throughout identity. The libs wrapper seems to have some retry / healthy endpoint selection logic

Reply:
we wanted to control the discovery node identity talks to, so I believe that's why it was manually used, should be fine either way